app/src/store/playground/playgroundStore.tsx

```diff
@@ -240,43 +244,76 @@ export const createPlaygroundStore = (initialProps: InitialPlaygroundState) => {
-            invocationParameters: [
-              ...instance.model.invocationParameters,
-              // These should never be changing at the same time as the provider but spread here to be safe
-              ...(model.invocationParameters ?? []),
-            ],
-          };
-        } else {
-          // Force clear the model name if the provider changes
-          newModel = {
-            ...newModel,
-            modelName: undefined,
-          };
-        }
+      if (instance.model.provider === provider) {
+        return;
       }
 
+      const savedProviderConfig = modelConfigByProvider[provider];
+
+      const patch: Partial<PlaygroundInstance> = {
+        // If we have a saved config for the provider, use it as the default otherwise reset the model config entirely to defaults / unset which will be controlled by invocation params coming from the server
+        model: savedProviderConfig
+          ? {
+              ...savedProviderConfig,
+              // Reset invocation parameters to unset, these will be subsequently fetched and updated from the server
+              // These are not be saved in the model config as they are controlled exclusively by the server
+              supportedInvocationParameters: [],
+              provider,
+            }
+          : {
+              modelName: null,
+              // Reset invocation parameters to unset, these will be subsequently fetched and updated from the server
+              invocationParameters: [],
+              // Reset supported invocation parameters to unset, these will be subsequently fetched and updated from the server
+              supportedInvocationParameters: [],
+              apiVersion: null,
+              endpoint: null,
+              provider,
+            },
+        tools: convertInstanceToolsToProvider({
+          instanceTools: instance.tools,
+          provider,
+        }),
+      };
+      if (instance.template.__type === "chat") {
+        patch.template = {
+          __type: "chat",
+          messages: instance.template.messages.map((message) => {
+            return {
+              ...message,
+              toolCalls: convertMessageToolCallsToProvider({
+                toolCalls: message.toolCalls,
+                provider,
+              }),
+            };
+          }),
+        };
+      }
+      set({
+        instances: instances.map((instance) => {
+          if (instance.id === instanceId) {
+            return {
+              ...instance,
+              ...patch,
+            };
+          }
+          return instance;
+        }),
+      });
+    },
+    updateModel: ({ instanceId, model }) => {
+      const instances = get().instances;
+      const instance = instances.find((instance) => instance.id === instanceId);
+      if (!instance) {
+        return;
+      }
+      const newModel = model;
```

Comment:
Could you rename this up in the destructured arguments

{ model: newModel }

It took me a minute to understand why the reassignment was happening here